controllers/studentController/index.ts

```diff
@@ -0,0 +1,12 @@
+import { Router } from 'express';
+import { createStudent , getAllStudents , loginStudent , registerStudent} from './studentController';
+import gatekeeper from '../../middleware/gatekeeper';
+
+const router = Router();
+
+router.post('/createStudent', gatekeeper(['admin']), createStudent);
```

Comment:
two routes have only gate keeper?  and are we implementing it here?

Reply:
it should be as general for all routes

Reply:
Done

Reply:
great solution. now lets add a rate limit for routes name. ye3ne heyde l route fya tkoun 5 marrat bl di2a w  tene route fya koun mrten bl di2a

Reply:
Done